packages/backend/src/transientGroupOwnershipResolver.ts

```diff
@@ -0,0 +1,91 @@
+import { DiscoveryService } from '@backstage/backend-plugin-api';
+import { CatalogApi, CatalogClient } from '@backstage/catalog-client';
+import {
+  Entity,
+  GroupEntity,
+  RELATION_CHILD_OF,
+  RELATION_MEMBER_OF,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { AuthOwnershipResolver } from '@backstage/plugin-auth-node';
+
+export class TransientGroupOwnershipResolver implements AuthOwnershipResolver {
+  private readonly catalogApi: CatalogApi;
+  private readonly config: Config;
+
+  constructor(deps: {
+    discovery: DiscoveryService;
+    config: Config;
+    catalog?: CatalogApi;
+  }) {
+    this.catalogApi = deps.catalog
+      ? deps.catalog
+      : new CatalogClient({ discoveryApi: deps.discovery });
+    this.config = deps.config;
+  }
+
+  private async resolveParentGroups(groupRefs: string[]): Promise<string[]> {
+    const allTransientGroupRefs = new Set<string>();
+
+    for (const groupRef of groupRefs) {
+      if (allTransientGroupRefs.has(groupRef)) continue;
+
+      allTransientGroupRefs.add(groupRef);
+
+      const res = await this.catalogApi.getEntitiesByRefs({
+        entityRefs: [groupRef],
+        fields: ['kind', 'relations'],
+      });
+
+      const groupEntity = res.items?.find(
+        e => e?.kind.toLowerCase() === 'group',
```

Comment:
I wonder if it would be safer to do this as `e?.kind.toLocaleLowerCase('en-US')}`?